src/inc/holder.h

```diff
@@ -13,8 +13,13 @@
 #include "staticcontract.h"
 #include "volatile.h"
 #include "palclr.h"
+#ifdef PAL_STDCPP_COMPAT
+#include <utility>
+#include <type_traits>
```

Comment:
Why do you need these ifdefs in these files under src\inc? I do not think they should be needed.